package/lib/SimpleSchema.js

```diff
@@ -353,18 +352,18 @@ class SimpleSchema {
       this._schema[key].type.definitions.forEach(typeDef => {
         if (!(SimpleSchema.isSimpleSchema(typeDef.type))) return;
         typeDef.type._blackboxKeys.forEach(blackboxKey => {
-          blackboxKeys.push(`${key}.${blackboxKey}`);
+          blackboxKeys.add(`${key}.${blackboxKey}`);
         });
       });
     });
-    return uniq(blackboxKeys);
+    return blackboxKeys;
```

Comment:
This needs to be `return Array.from(blackboxKeys);`. Code that uses this expects it to return an array.

Also, I think I see an issue here, which might be the actual performance issue you were seeing rather than array vs. set. `const blackboxKeys = this._blackboxKeys;` doesn't clone the array (or the set); it only gets a reference to it. So every time this function runs, it's pushing more keys into the same array.

Other than a performance issue, it's also wrong. `this._blackboxKeys` isn't supposed to include the subschema keys and now it will after the first time this runs. This is still a problem even after your changes.

So at minimum the first line should be cloning the set: `const blackboxKeys = new Set(this._blackboxKeys);`

But I think you will also need to change

```js
typeDef.type._blackboxKeys.forEach
```

to

```js
typeDef.type.blackboxKeys().forEach
```

So that all of the levels of schemas properly "bubble up"

Reply:
Your suggestions are now pushed. Thanks to your comment, I understand better how blackboxkeys should be treated. Each level of schemas must contain only its own level of blackboxkeys and the method must return the aggregation of the current level with the levels below.
The tests pass now, sorry for the regression.